packages/zowe-explorer/src/uss/utils.ts

```diff
@@ -78,19 +70,19 @@ export function disposeClipboardContents(): void {
 }
 
 export async function autoDetectEncoding(node: IZoweUSSTreeNode, profile?: imperative.IProfileLoaded): Promise<void> {
-    if (node.binary || node.encoding !== undefined) {
+    if (node.getEncoding() !== undefined) {
         return;
     }
     const ussApi = ZoweExplorerApiRegister.getUssApi(profile ?? node.getProfile());
     if (ussApi.getTag != null) {
         const taggedEncoding = await ussApi.getTag(node.fullPath);
         if (taggedEncoding === "binary" || taggedEncoding === "mixed") {
-            node.setEncoding({ kind: "binary" });
-        } else {
-            node.setEncoding(taggedEncoding !== "untagged" ? { kind: "other", codepage: taggedEncoding } : undefined);
+            await node.setEncoding({ kind: "binary" });
+        } else if (taggedEncoding !== "untagged") {
+            await node.setEncoding({ kind: "other", codepage: taggedEncoding });
         }
     } else {
         const isBinary = await ussApi.isFileTagBinOrAscii(node.fullPath);
-        node.setEncoding(isBinary ? { kind: "binary" } : undefined);
+        await node.setEncoding(isBinary ? { kind: "binary" } : undefined);
```

Comment:
It seems that we have special behavior defined for `encoding === null` based on this:
- `if (node.getEncoding() !== undefined) { return }`

IIRC, it means that we want to go with the default z/OS (MF/RSE/FTP) encoding.
> Thanks @t1m0thyj for confirming, I won't forget it again ([**_source_**](https://github.com/zowe/zowe-explorer-vscode/blob/206bf062390e8b58cd6364e485a4fe21ce4ba1f5/packages/zowe-explorer-api/src/tree/IZoweTreeNode.ts#L182-L187))

Shouldn't we set the encoding to `null` instead of `undefined` if it's not tagged and is not binary?
Otherwise, I can see how we could be calling the `autoDetectEncoding` method in an infinite loop trying to determine the encoding but never getting anywhere because we are the ones setting it to `undefined` 😋 

Please let me know if this makes no sense 😅